lib/Driver/ToolChains.cpp

```diff
@@ -1112,6 +1112,7 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
         Twine("@") + OutputInfo.getPrimaryOutputFilename()));
   }
 
+#if !defined(__CYGWIN__)
```

Comment:
We don't use `#if` in the compiler.  Please create a separate toolchain class for Cygwin, if it requires customization, or create customization parameters in the `GenericUnix` constructor.